### PR TITLE
Fixed some issues with the leaflet map.

### PR DESCRIPTION
**Pull request by Hugovs2000**
Thought of a better way of animating the lap by moving the circle instead of reprinting it each time as well as adding coordinates to the polyline instead of redrawing the polyline with an updated array of lats and longs.

### CODE REVIEW

File: src/dom/map-setup.ts [hunk 1]
```diff
@@ -78,10 +57,8 @@ export function animateLap(lapJSON: ILapDataset) {
     let long = lapJSON.dataSet[counter]["Lon."] * 0.000001;
     let lat = lapJSON.dataSet[counter]["Lat."] * 0.000001;
```
Comment (Musa-Mabasa) [inline src/dom/map-setup.ts:59]: Did you perform safely checks for these values? If so then no issues.

**Review by LesediMoko — APPROVED** [review]
Besides the possible left-out safety checks that Musa has commented on, everything looks really good. Nice work!

File: src/dom/map-setup.ts (hunk 1 above)
Comment (Hugovs2000) [reply src/dom/map-setup.ts:59]: I do right above it. It is just cut off:
```
if (
      !(
        lapJSON?.dataSet?.[counter]?.["Lat."] ||
        lapJSON?.dataSet?.[counter]?.["Lon."]
      )
    ) {
      return;
    }
```

Comment (Musa-Mabasa) [reply src/dom/map-setup.ts:59]: Ohh I see, it wasn't part of the changes.

**Review by Musa-Mabasa — APPROVED** [review]
Everything looks good 👍

**Pull request merged**